### ISSUE TITLE
[Feature Request] Toggle blur (On/Off)

### ISSUE THREAD

**Issue by RealAzmodeo** [zanllp/sd-webui-infinite-image-browsing#279]
I think this is one of the most useful complements for A1111 and I use it a lot. The thing is that sometimes you get NSFW images and you want to keep them, but you don't want other people to see them (duh), so it would be great to either have a "hide" option for images or a blur out one.

**Comment by zanllp** [comment]
Your idea sounds good, but the challenge lies in how to determine whether an image is NSFW or not. The simplest method would be manual labeling, but that seems a bit cumbersome. There are automated solutions available, such as image recognition algorithms, but they are not 100% accurate and may produce false positives or negatives. It's a tricky problem that requires careful consideration and testing to find the best solution.

**Comment by RealAzmodeo** [comment]
As a matter of fact I think the best way possible would be to do it manually. The plugin already support batch selection in case you want to do that. Also, by doing it manually you keep users agency over what they want to be shown and what they want to keep blured.

**Comment by zanllp** [comment]
It seems like implementing this feature would require significant changes to the current system. Therefore, I will prioritize other higher priority requirements first and revisit this suggestion at a later time to explore how we can implement it effectively. 

However, I do agree that a "hide" or blur option for NSFW images would be useful and will keep it in mind for future updates. Thank you for your input!

**Comment by WSH032** [comment]
> Your idea sounds good, but the challenge lies in how to determine whether an image is NSFW or not. The simplest method would be manual labeling, but that seems a bit cumbersome. There are automated solutions available, such as image recognition algorithms, but they are not 100% accurate and may produce false positives or negatives. It's a tricky problem that requires careful consideration and testing to find the best solution.

I think we can rely on the tag keyword in exif to try and determine,
Set up a vocabulary, such as `nsfw, <some sensitive tag>...`
If any of these words appear in exif, it will be marked as nsfw

---

For images without exif, as far as I know, with GPU support, smilewolf's WD14 model can effectively identify sensitive images of anime characters

Of course, annotating a large number of images can consume a lot of time,  may not be very recommended

**Comment by RealAzmodeo** [comment]
thanks for considering this. I still think that providing control for every user to select which image they want to blur is much better than the system doing it automatically. I mean, maybe that can also be an option and users can have this setting on so it automatically decides if it is NSFW or not, but they should ever have control on what they want to show here.
Now another option is to have a toggle to hide images NSFW completely. That would be a good option as well.

**Comment by zanllp** [comment]
Yes, perhaps we can add a special custom tag, such as "NSFW". Then you can manually add this tag to images, and when I detect that some images have this special tag, I will add a blurred mask to them. There are too many unfinished feature requests in this repo, and I can't implement them all. This is a relatively easy way to implement this feature.

**Comment by zanllp** [comment]
By the way, batch tagging functionality will likely be added later.

**Comment by RealAzmodeo** [comment]
> By the way, batch tagging functionality will likely be added later.

Woooooow. Nice
